src/commands/install.yml

```diff
@@ -31,27 +31,32 @@ steps:
   - run:
       name: "Install Go"
       command: |
-        if which go;then
-          if go version | grep "<< parameters.version >>";then
+        if command -v go &>/dev/null;then
+          if go version | grep -q -F "go<< parameters.version >> ";then
             echo "Binary already exists, skipping download."
             exit 0
           else
             echo "Go is already install but is the wrong version."
             echo "Installing the request version instead."
           fi
 
-          echo "Installing the requested version of Go."
-        fi
+          GOHOSTARCH="$(go env GOHOSTARCH)"
+          GOHOSTOS="$(go env GOHOSTOS)"
 
-        if [[ $OSD_FAMILY == "linux" ]];then
-          curl -sSL "https://dl.google.com/go/go<< parameters.version >>.linux-amd64.tar.gz" | sudo tar -xz -C /usr/local/
-        elif [[ $OSD_FAMILY == "darwin" ]];then
-          curl -sSL "https://dl.google.com/go/go<< parameters.version >>.darwin-amd64.tar.gz" | sudo tar -xz -C /usr/local/
+          $SUDO rm -rf /usr/local/go &>/dev/null
+          $SUDO install --owner=${USER} -d /usr/local/go
         else
-          echo "OS/platform not supported."
-          exit 1
+          echo "Installing the requested version of Go."
+
+          GOHOSTARCH="${HOSTTYPE}"
+          if [[ "${GOHOSTARCH}" == "x86_64" ]]; then GOHOSTARCH="amd64"; fi
+          if [[ "${GOHOSTARCH}" == *"86" ]]; then GOHOSTARCH="i386"; fi
```

Comment:
Would prefer to find a POSIX-complaint way of doing this for better compatibility.

Reply:
As we set `GOHOSTARCH` ourselves, that's probably not the POSIX non-compliance you have in mind.

Unless I'm mistaken the Orb(s) utilize *bash*. In general, all are riddled with `echo` instead of `printf`, so the ship has sailed long ago for the entire collection. :-)

Reply:
Orbs will utilise whatever the default shell of the job is - `[[ ]]` and `==` are not POSIX compliant, and won't work in alternative shells like `ash` in Alpine. We default to `bash` in our own images, but given we're attempting to add more flexibility to the script, it may be worth adding compatibility for arch.